docs/report_identification.md

```diff
@@ -0,0 +1,28 @@
+# Analyzer report identification
+
+Unique report idetifiers are required to compare analysis results.
+If the static analyzers (Clang Static Analyzer, Clang Tidy) do not generate
+a unique report identifier, CodeChecker tries to generate a unique id.
+
+It is recommended to use the identifier generated by the static analyzers
+because they can include the semantic context better of the result into the
+unique id.
+
+## Experimental context free hash generation method
+
+This method is used for Clang Tidy reports.
+The current hash generation algrithm in CodeChecker uses this information to
+generate the unique id.
+__NOTE: As the main diagnositc section the last element from the bug path is used!__
+
+Unique id content:
+
+  * `file_name` from the main diag section.
+  * `checker name`
+  * `checker message`
+  * `line content` from the source file if can be read up
```

Comment:
Is it possible that we cannot read up the source file?

Reply:
Yes if there is some encoding problem.